### PR TITLE
Rework date picker and field selection

### PR DESCRIPTION
**Pull request by ZacSweers**
This reworks the date picker in a few ways
- Full date picker uses a better implementation from Calf to show a a native date picker UI per-platform.
- Show the date picker in a bottom sheet instead of a dialog. Mostly because of a lack of a flexible adaptive dialog on iOS

This also makes field picking a little more structured with two hierarchical dropdowns

Problems
- [x] Compose Material3's bottom sheet APIs remain bad and weird to use
- [x] iOS UIDatePicker shows as... black? For some reason.
- [x] iOS date picker doesn't reflect dark mode or support color params
- [x] iOS Date Picker doesn't automatically change to reflect dark mode changes
- [ ] You can drag the calendar on iOS ever so slightly and reveal a white background on dark mode. Idk what to do about this

Tbh will probably move to just have a days tab pager similar to a calendar widget

### CODE REVIEW

File: src/commonMain/kotlin/dev/zacsweers/fieldspottr/Home.kt [hunk 1]
```diff
@@ -216,27 +220,27 @@ fun Home(state: HomeScreen.State, modifier: Modifier = Modifier) {
...
-        state.selectedGroup,
-        modifier = Modifier.align(CenterHorizontally).padding(horizontal = 16.dp),
-      ) { newGroup ->
+    Column(modifier = Modifier.padding(innerPadding), verticalArrangement = spacedBy(8.dp)) {
+      GroupSelector(state.selectedGroup, modifier = Modifier.align(CenterHorizontally)) { newGroup
+        ->
         state.eventSink(ChangeGroup(newGroup))
       }
 
+      val cornerSlot = movableContentOf {
```
Comment (alexvanyo) [inline src/commonMain/kotlin/dev/zacsweers/fieldspottr/Home.kt:235]: A `movableContentOf` that isn't `remember`ed is suspicious here, that could lead to the state loss issue inside `DateSelector` if this re-runs and a new `movableContentOf` is created.

Comment (ZacSweers) [reply src/commonMain/kotlin/dev/zacsweers/fieldspottr/Home.kt:235]: oooo that's a good catch!

Comment (ZacSweers) [reply src/commonMain/kotlin/dev/zacsweers/fieldspottr/Home.kt]: [0e828dd](https://github.com/ZacSweers/FieldSpottr/pull/26/commits/0e828dd3e35586a6833f08400429ae29f1a14309)

**Comment by MohamedRejeb** [comment]
I encountered revealing the bottom sheet bg color issue on iOS. I fixed it on Calf by setting the container color to UIKit sheet bg color. It will be fixed in the next release.

**Comment by ZacSweers** [comment]
Tested in 0.5.4 and working great!

**Pull request merged**